api/client.go

```diff
@@ -95,7 +96,11 @@ func (c *Client) do(ctx context.Context, method, path string, reqData, respData
 	var reqBody io.Reader
 	var data []byte
 	var err error
-	if reqData != nil {
+	switch reqData := reqData.(type) {
+	case io.Reader:
+		reqBody = reqData
```

Comment:
```suggestion
        // uploading a file on disc
		reqBody = reqData
```

Reply:
Added a comment. However it's not necessarily a file on disk but can be any io.Reader